### ISSUE TITLE
🐛 `noUnusedVariables` makes the semantic model panics on invalid code

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#4114]
### Environment information

```block
Main
```

### What happened?

Given the following file named `main,js`:

```js
c;
const c = 0;
```

Execute the following command:

```shell
biome lint --only=correctness/noUnusedVariables main.js
````

And then notice the panics:

```shell
Biome encountered an unexpected error

This is a bug in Biome, not an error in your code, and we would appreciate it if you could report it to https://github.com/biomejs/biome/issues/ along with the following information to help us fixing the issue:

Source Location: crates/biome_js_semantic/src/semantic_model/model.rs:138:21
Thread Name: biome::worker_4
Message: internal error: entered unreachable code: Expected global scope not present

main.js internalError/panic  INTERNAL  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ processing panicked: internal error: entered unreachable code: Expected global scope not present
  
  ⚠ This diagnostic was derived from an internal Biome error. Potential bug, please report it if necessary.
  

Checked 0 files in 1558µs. No fixes applied.
internalError/io ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ No files were processed in the specified paths.
```

### Expected result

Biome should not panic.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
